tests/test_autograder.py

```diff
@@ -31,45 +32,46 @@
         arm=False,
     )
 
-    ag.create_public_io_tests(
-        [
-            ("1", 1),
-            ("2", 2),
-        ],
-        prefix="Public Tests",
-    )
-
-    ag.create_private_io_tests(
-        [
-            ("3", 4),
-            ("4", 4),
-        ],
-        prefix="Hidden Tests",
-    )
-
     def test_num_lines(result: CustomTestCaseResult):
         line_count = 0
         with open("./palindrome.c", "r") as fp:
             line_count = len(fp.readlines())
-        
+
         if line_count > 20:
             result.passed = False
             result.output = "You wrote too much code!"
-            result.score = 0 
+            result.score = 0
         else:
             result.passed = True
             result.output = "Line number within limit."
             result.score = 4
 
-    custom_test = CustomTestCase(
-        test_num_lines,
-        "Custom Test Case 1",
-        point_value = 4,
-        visibility=Visibility.VISIBLE,
-    )
+    custom_test = CustomTestCase(test_num_lines, "Custom Test Case 1", point_value=4)
 
     ag.add_test(custom_test)
 
+    ag.io_tests_bulk_loader(
+        prefix="Unit Tests - ",
+        default_timeout_ms=5000,
+        commands_prefix="cmd",
+        test_input_prefix="test",
+        expected_stderr_prefix="err",
+        expected_stdout_prefix="out",
+    ).add(
+        "1",
+        2,
+        timeout_ms=20000,
+        prefix="Public - ",
+    ).add_list(
+        [
+            ("2", 4),
+            ("3", 4),
+            ("4", 4),
+        ],
+        prefix="Hidden - ",
+        hidden=True,
+    )
+
```

Comment:
@DuckDuckWhaleUCSD This is the file I mentioned where I test the up-to-date autograder implementation.

Also lemme know what you think of this new interface for bulk loading test cases. I wrote it so you can chain methods like this but you don't have to write it like this of course...